### ISSUE TITLE
the loss is strange

### ISSUE THREAD

**Issue by zhouweii234** [hkchengrex/STCN#28]
The stage 0 total_loss is very strange. The model seems to be overfitting. But the model converges at about 20k step, this is even earlier than when the model was first saved (50k step). Since I didn't change any parameter except `master_port` and I also trained the model with two 11GB 2080Ti GPUs as you write in the article, the time that the model converges seems to be too early. Is this what the loss supposed to be like? Or I should use the model that just converges (when the loss is lowest) to train the stage 3?
![lossstage0](https://user-images.githubusercontent.com/62086490/126324614-03f4c132-ebd6-44c3-a440-8b408d50e54c.png)

By the way, the total_loss of stage 3 is rising when I use the final model of stage 0 (at 300k step) to train it. Is the loss of stage 3 supposed to rise? Or this is just the wrong result of over fitting stage 0 model?
![lossstage3](https://user-images.githubusercontent.com/62086490/126325508-32b6581c-4d3c-45ca-969f-0045861166ad.png)

**Comment by hkchengrex** [comment]
The rise after some iterations is normal because we are using bootstrapped CE loss: https://github.com/hkchengrex/STCN/blob/main/model/losses.py
The loss for stage 3 is also not that straightforward because we are using curriculum learning, i.e., the difficulty of training samples increases over time.
